packages/browser/src/types/analytics.ts

```diff
@@ -26,41 +26,70 @@ export interface ExperimentAnalyticsEvent {
    * User properties to identify with the user prior to sending the event.
    */
   userProperties?: Record<string, unknown>;
-}
-
-/**
- * Event for tracking a user's exposure to a variant. This event will not count
- * towards your analytics event volume.
- */
-export class ExposureEvent implements ExperimentAnalyticsEvent {
```

Comment:
Is this pattern not common in js-land? In other words is it more "native" to use a function to return an interface implementation?

Reply:
for simple data objections, factory functions are preferable to classes